pandas/_libs/lib.pyx

```diff
@@ -642,6 +643,39 @@ def array_equivalent_object(ndarray left, ndarray right) -> bool:
     return True
 
 
+ctypedef fused int6432_t:
+    int64_t
+    int32_t
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def array_equal_fast(
+    ndarray[int6432_t, ndim=1] left, ndarray[int6432_t, ndim=1] right,
+) -> bool:
+    """
+    Perform an element by element comparison on 1-d integer arrays, meant for indexer
+    comparisons
+    """
+    cdef:
+        Py_ssize_t i, n = left.size
+
+    if left.size != right.size:
+        return False
+
+    with nogil:
+        for i in range(n):
+
+            if left[i] != right[i]:
+                break
```

Comment:
Why not return here?

Reply:
Can’t return python values without the gil

Reply:
Hmm, is there a reason to release the gil here?

IIUC, perf shouldn't be hurt by holding on to the gil here, and this function is not declared as cpdef(so it doesn't seem like other nogil functions in _libs would call it).

Reply:
Hm good point. I had the same assumption but this ran significantly slower with the Gil initially, re-ran the timing now and it's equally fast now. Must have been something weird on my machine